src/Microsoft.Health.Dicom.Functions.UnitTests/Management/PurgeOrchestrationInstancesHistoryTests.cs

```diff
@@ -0,0 +1,132 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Management;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Functions.UnitTests
+{
+    public class PurgeOrchestrationInstancesHistoryTests
+    {
+        private readonly PurgeOrchestrationInstancesHistoryConfiguration _purgeConfig;
+        private readonly TimerInfo _timer;
+        private readonly ILogger _logger;
+        private readonly DateTime _definedNow;
+        private readonly PurgeOrchestrationInstancesHistory _purgeOrchestrationInstancesHistory;
+        private readonly IDurableOrchestrationClient _durableOrchestrationClientMock;
+
+        public PurgeOrchestrationInstancesHistoryTests()
+        {
+            _purgeConfig = new PurgeOrchestrationInstancesHistoryConfiguration();
+            _purgeConfig.RuntimeStatuses = new OrchestrationRuntimeStatus[] { OrchestrationRuntimeStatus.Completed };
+            _timer = Substitute.For<TimerInfo>(default, default, default);
+            _logger = Substitute.For<ILogger>();
+            _definedNow = DateTime.UtcNow;
+            _purgeOrchestrationInstancesHistory = new PurgeOrchestrationInstancesHistory(
+                Options.Create(_purgeConfig),
+                () => _definedNow);
+            _durableOrchestrationClientMock = Substitute.For<IDurableOrchestrationClient>();
+        }
+
+        [Fact]
+        public async Task GivenValidInput_PurgeCompletedDurableFunctionsHistory_ShouldNotPurgeAnythingOrchestrationsAsync()
+        {
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(givenCondition => IsCondition(givenCondition, null)),
+                    Arg.Is<CancellationToken>(cancellationTokenSource.Token))
+                .Returns(Task.FromResult(new OrchestrationStatusQueryResult { DurableOrchestrationState = Array.Empty<DurableOrchestrationStatus>() }));
+
+            await _purgeOrchestrationInstancesHistory.Run(_timer, _durableOrchestrationClientMock, _logger, cancellationTokenSource.Token);
+
+            await _durableOrchestrationClientMock
+                .DidNotReceiveWithAnyArgs()
+                .PurgeInstanceHistoryAsync(Arg.Any<string>());
+        }
+
+        [Fact]
+        public async Task GivenValidInput_PurgeCompletedDurableFunctionsHistory_ShouldPurgeMultiPageListOfOrchestrationsAsync()
+        {
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+            List<DurableOrchestrationStatus> durableOrchestrationStatuses = new List<DurableOrchestrationStatus>()
```

Comment:
nit: be consistent with `var` with its usage below. Pick one style or the other 😛